src/EA.Weee.RequestHandlers/Mappings/SmallProducerSubmissionHistoryDataMap.cs

```diff
@@ -0,0 +1,91 @@
+﻿namespace EA.Weee.RequestHandlers.Mappings
+{
+    using Core.DirectRegistrant;
+    using CuttingEdge.Conditions;
+    using Domain.Organisation;
+    using Domain.Producer;
+    using EA.Prsd.Core.Mapper;
+    using EA.Weee.Core.DataReturns;
+    using EA.Weee.Core.Organisations;
+    using EA.Weee.Core.Shared;
+    using EA.Weee.Domain.DataReturns;
+    using System.Collections.Generic;
+
+    internal class SmallProducerSubmissionHistoryDataMap : IMap<DirectProducerSubmissionSource, SmallProducerSubmissionHistoryData>
+    {
+        private readonly IMapper mapper;
+
+        public SmallProducerSubmissionHistoryDataMap(IMapper mapper)
+        {
+            this.mapper = mapper;
+        }
+
+        public SmallProducerSubmissionHistoryData Map(DirectProducerSubmissionSource source)
+        {
+            Condition.Requires(source).IsNotNull();
+
+            return new SmallProducerSubmissionHistoryData()
```

Comment:
@JohnPadgettCivica so many short hand conditions here makes it difficult to read. It's minor, but is there not a way to check once if CurrentSubmission is null/ empty (by allowing it to be nullable further up) then build for DirectRegistrant.Organisation else build for DirectProducerSubmission.CurrentSubmission

Reply:
@seanciv Updated to make more readable